youtube_downloader/downloader.py

```
@@ -43,8 +97,45 @@ def _get_video_info(self):
+                if response.status_code == 429:
+                    if self.proxy_manager:
+                        print("⚠ Rate limited (429). Rotating proxy...")
+                        if current_proxy:
+                            self.proxy_manager.record_failure(
+                                current_proxy,
+                                Exception("429 Too Many Requests")
+                            )
+                        self._rotate_proxy()
+                        if attempt < retries - 1:
+                            continue
+                    raise Exception("Rate limited by YouTube (429 Too Many Requests)")
```

Ruff (0.14.1)

Create your own exception

(TRY002)